File: scratch/refactor.js
const fs = require('fs');

const appPath = 'src/App.tsx';
let content = fs.readFileSync(appPath, 'utf8');

const importStr = 'import { formatTime, timeAgo } from "./ui/utils";';
if (content.includes(importStr)) {
  content = content.replace(importStr, importStr + '\nimport {\n  Artwork,\n  BrandHeader,\n  Action,\n  SectionHeader,\n  ScreenTitle,\n  TrackRow as TrackLine,\n  SectionRail,\n  MiniPlayer,\n} from "./ui/components";');
}

const artworkStart = content.indexOf('function Artwork({');
const mixPanelStart = content.indexOf('function DiscoverMixPanel({');

if (artworkStart !== -1 && mixPanelStart !== -1) {
  content = content.substring(0, artworkStart) + content.substring(mixPanelStart);
}

const miniPlayerStart = content.indexOf('function MiniPlayer({');
const bottomNavStart = content.indexOf('function BottomNav({');

if (miniPlayerStart !== -1 && bottomNavStart !== -1) {
  content = content.substring(0, miniPlayerStart) + content.substring(bottomNavStart);
}

fs.writeFileSync(appPath, content, 'utf8');
console.log('Refactored App.tsx');

File: scratch/refactor.py
import os

app_path = "src/App.tsx"
with open(app_path, "r", encoding="utf-8") as f:
    lines = f.readlines()

# Find the import
import_idx = -1
for i, line in enumerate(lines):
    if 'import { formatTime, timeAgo } from "./ui/utils";' in line:
        import_idx = i
        break

if import_idx != -1:
    imports_to_add = """import {
  Artwork,
  BrandHeader,
  Action,
  SectionHeader,
  ScreenTitle,
  TrackRow as TrackLine,
  SectionRail,
  MiniPlayer,
} from "./ui/components";\n"""
    lines.insert(import_idx + 1, imports_to_add)

# Find Artwork start and SectionRail end
artwork_start = -1
section_rail_end = -1
for i, line in enumerate(lines):
    if line.startswith("function Artwork({"):
        if artwork_start == -1:
            artwork_start = i
    if line.startswith("function DiscoverMixPanel({"):
        if section_rail_end == -1:
            section_rail_end = i - 1

if artwork_start != -1 and section_rail_end != -1:
    # Delete lines
    del lines[artwork_start:section_rail_end+1]

# Now find MiniPlayer start and BottomNav start
mini_player_start = -1
bottom_nav_start = -1
for i, line in enumerate(lines):
    if line.startswith("function MiniPlayer({"):
        mini_player_start = i
    if line.startswith("function BottomNav({"):
        bottom_nav_start = i - 1

if mini_player_start != -1 and bottom_nav_start != -1:
    del lines[mini_player_start:bottom_nav_start+1]

with open(app_path, "w", encoding="utf-8") as f:
    f.writelines(lines)

print("Refactored App.tsx")
